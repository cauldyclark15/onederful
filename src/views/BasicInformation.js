import React from 'react';
import PropTypes from 'prop-types';
import { Card, Table } from 'antd';

import BasicInfoRow from './partials/BasicInfoRow';

const propTypes = {
  name: PropTypes.string,
  address: PropTypes.string,
  member_id: PropTypes.string,
  dob: PropTypes.string,
  gender: PropTypes.string,
  reference: PropTypes.object,
  date_time_periods: PropTypes.array,
  entity: PropTypes.string,
  zip_code: PropTypes.string,
};

const defaultProps = {
  name: 'User',
  address: 'n/a',
  member_id: 'n/a',
  dob: 'n/a',
  gender: 'n/a',
  reference: { description: 'n/a' },
  date_time_periods: [],
  entity: 'n/a',
  zip_code: 'n/a',
};

function BasicInformation({
  name,
  address,
  member_id,
  dob,
  gender,
  reference,
  date_time_periods,
  entity,
  zip_code,
}) {
  const data = date_time_periods.map((dtp, index) => ({ ...dtp, key: index }));
  const columns = [
    {
      title: 'Date',
      dataIndex: 'date_string',
    },
    {
      title: 'Qualifier',
      dataIndex: 'date_qualifier',
    },
  ];

  return (
    <Card title={name} style={{ width: '100%', height: '100%' }}>
      <BasicInfoRow label="Address" value={address} />
      <BasicInfoRow label="Member ID" value={member_id} />
      <BasicInfoRow label="Date of birth" value={dob} />
      <BasicInfoRow label="Gender" value={gender} />
      <BasicInfoRow label="Reference" value={reference.description} />
      <BasicInfoRow label="Entity" value={entity} />
      <BasicInfoRow label="Zip code" value={zip_code} />
      <h3>Date time periods</h3>
      <Table columns={columns} dataSource={data} />
    </Card>
  );
}

BasicInformation.propTypes = propTypes;
BasicInformation.defaultProps = defaultProps;

export default BasicInformation;
