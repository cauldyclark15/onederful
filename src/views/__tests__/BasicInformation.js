import React from 'react';
import renderer from 'react-test-renderer';

import BasicInformation from '../BasicInformation';

it('renders correctly and unchange so far', () => {
  const tree = renderer.create(<BasicInformation />).toJSON();

  expect(tree).toMatchSnapshot();
});
