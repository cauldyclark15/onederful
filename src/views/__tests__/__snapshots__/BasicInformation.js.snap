// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and unchange so far 1`] = `
<div
  className="ant-card ant-card-bordered"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        User
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Address
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Member ID
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Date of birth
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Gender
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Reference
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Entity
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-6"
      >
        <h4>
          Zip code
        </h4>
      </div>
      <div
        className="ant-col-18"
      >
        <p>
          n/a
        </p>
      </div>
    </div>
    <h3>
      Date time periods
    </h3>
    <div
      className="ant-table-wrapper"
    >
      <div
        className="ant-spin-nested-loading"
        style={null}
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
            style={Object {}}
          >
            <div
              className="ant-table-content"
            >
              <div
                className="ant-table-body"
                onScroll={[Function]}
                onWheel={[Function]}
                style={Object {}}
              >
                <table
                  className=""
                  style={Object {}}
                >
                  <colgroup>
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr
                      style={
                        Object {
                          "height": null,
                        }
                      }
                    >
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          Date
                        </div>
                      </th>
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          Qualifier
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  />
                </table>
              </div>
              <div
                className="ant-table-placeholder"
              >
                No data
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
